### PR TITLE
Use MimeTypes package to determine MIME type

### PR DESCRIPTION
**Pull request by holahmeds**
This simplifies the code since we don't have to keep large mappings of extensions and MIME types. The current method is also reportedly error prone (#494).

We still keep the ability to override the mappings for:
- filling in entries not present in the package, for e.g. ".azw3"
- picking preferred extensions, for e.g. MimeTypes provides ".conf" as a possible extionsion for "text/plain", and while that is correct, ".txt" would be preferrable
- compatibility reasons

**Issues**
#494

Library: https://github.com/khellang/MimeTypes

### CODE REVIEW

File: MediaBrowser.Model/Net/MimeTypes.cs [hunk 1]
```diff
@@ -231,6 +176,12 @@ public static class MimeTypes
                 return result;
             }
 
+            var extensions = Model.MimeTypes.GetMimeTypeExtensions(mimeType);
+            if (extensions.Any())
+            {
+                return "." + extensions.First();
+            }
+
             return null;
```
Comment (crobibero) [inline MediaBrowser.Model/Net/MimeTypes.cs:185]:
```suggestion
            var extension = Model.MimeTypes.GetMimeTypeExtensions(mimeType).FirstOrDefault();
            return string.IsNullOrEmpty(extension) ? null : "." + extension;
```

File: MediaBrowser.Model/Net/MimeTypes.cs [hunk 2]
```diff
@@ -12,6 +12,10 @@ namespace MediaBrowser.Model.Net
     /// <summary>
     /// Class MimeTypes.
     /// </summary>
+    ///
+    /// http://en.wikipedia.org/wiki/Internet_media_type
```
Comment (cvium) [inline MediaBrowser.Model/Net/MimeTypes.cs:16]: Maybe this should be put in a `<remarks>` tag?

File: tests/Jellyfin.Model.Tests/Net/MimeTypesTests.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        [InlineData(".oga", "audio/ogg")]
+        [InlineData(".ogg", "audio/ogg")]
+        [InlineData(".opus", "audio/ogg")]
+        [InlineData(".vorbis", "audio/vorbis")]
+        [InlineData(".wav", "audio/wav")]
+        [InlineData(".webma", "audio/webm")]
+        [InlineData(".wma", "audio/x-ms-wma")]
+        [InlineData(".wv", "audio/x-wavpack")]
+        [InlineData(".xsp", "audio/xsp")]
+        public void GetMimeType(string input, string expectedResult)
```
Comment (cvium) [inline tests/Jellyfin.Model.Tests/Net/MimeTypesTests.cs:90]: We usually follow the `MethodName_State_Expectation` naming scheme.
```suggestion
        public void GetMimeType_Valid_ReturnsCorrectResult(string input, string expectedResult)
```

File: tests/Jellyfin.Model.Tests/Net/MimeTypesTests.cs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        [InlineData("video/quicktime", ".mov")]
+        [InlineData("video/vnd.mpeg.dash.mpd", ".mpd")]
+        [InlineData("video/webm", ".webm")]
+        [InlineData("video/x-flv", ".flv")]
+        [InlineData("video/x-m4v", ".m4v")]
+        [InlineData("video/x-matroska", ".mkv")]
+        [InlineData("video/x-ms-asf", ".asf")]
+        [InlineData("video/x-ms-wmv", ".wmv")]
+        [InlineData("video/x-msvideo", ".avi")]
+        public void ToExtension(string input, string expectedResult)
```
Comment (cvium) [inline tests/Jellyfin.Model.Tests/Net/MimeTypesTests.cs:159]:
```suggestion
        public void ToExtension_Valid_ReturnsCorrectResult(string input, string expectedResult)
```

**Review by crobibero — APPROVED** [review]

File: MediaBrowser.Model/Net/MimeTypes.cs (hunk 2 above)
Comment (holahmeds) [reply MediaBrowser.Model/Net/MimeTypes.cs]: Updated

**Review by cvium — APPROVED** [review]

**Review by 1337joe — APPROVED** [review]

**Pull request merged**